metadata-integration/java/datahub-client/src/main/java/datahub/client/patch/common/CustomPropertiesPatchBuilder.java

```diff
@@ -0,0 +1,61 @@
+package datahub.client.patch.common;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import datahub.client.patch.AbstractMultiFieldPatchBuilder;
+import datahub.client.patch.PatchOperationType;
+import datahub.client.patch.subtypesupport.IntermediatePatchBuilder;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.lang3.tuple.ImmutableTriple;
+
+import static com.fasterxml.jackson.databind.node.JsonNodeFactory.*;
+
+
+public class CustomPropertiesPatchBuilder<T extends AbstractMultiFieldPatchBuilder<T>> implements IntermediatePatchBuilder<T> {
+
+  public static final String CUSTOM_PROPERTIES_BASE_PATH = "/customProperties/";
+
+  private T parent;
+  private final List<ImmutableTriple<String, String, JsonNode>> operations = new ArrayList<>();
+
+  public CustomPropertiesPatchBuilder(T parentBuilder) {
+    this.parent = parentBuilder;
+  }
+
+  /**
+   * Add a property to a custom properties field
+   * @param key
+   * @param value
+   * @return
+   */
+  public CustomPropertiesPatchBuilder<T> addProperty(String key, String value) {
+    operations.add(ImmutableTriple.of(PatchOperationType.ADD.getValue(), CUSTOM_PROPERTIES_BASE_PATH + key,
+        instance.textNode(value)));
+    return this;
+  }
+
+  /**
+   * Remove a property from a custom properties field. If the property doesn't exist, this is a no-op.
+   * @param key
+   * @return
+   */
+  public CustomPropertiesPatchBuilder<T> removeProperty(String key) {
+    return removeProperty(key, null);
+  }
+
+  private CustomPropertiesPatchBuilder<T> removeProperty(String key, String value) {
```

Comment:
@RyanHolstien : curious why you created removeProperty with key and value both needing to get specified? Isn't `removeProperty(String key)` more natural?

Reply:
Yeah this doesn't really need to be split out to a separate method, I think when I was writing it I thought this was more readable, but it's kind of less clear. I'm okay with changing it to just be in the public method